### PR TITLE
feat: Add implementation and configuration classes for OpenSearch vector storage

### PR DESCRIPTION
**Pull request by fuyou-lxm**
Describe what this PR does / why we need it
Add implementation and configuration classes for OpenSearch vector storage
Does this pull request fix one issue?
yes,https://github.com/alibaba/spring-ai-alibaba/issues/230
Describe how you did it
- Added OpenSearchVector.java file, which implements the VectorStore interface for interacting with Alibaba OpenSearch. This class supports functionalities such as adding, deleting, and similarity searching of vector documents.
- Added OpenSearchConfig.java file, which defines the parameters required to connect and configure OpenSearch, including instance ID, endpoint, username, and password.

Describe how to verify it

Special notes for reviews

### CODE REVIEW

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	public String getAccessUserName() { // 新增
+		return accessUserName;
+	}
+
+	public void setAccessUserName(String accessUserName) { // 新增
+		this.accessUserName = accessUserName;
+	}
+
+	public String getAccessPassWord() { // 新增
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java:51]: 注释用英语吧

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+		}
+		catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Represents the response body from a search operation.
+	 *
+	 * @author fuyou.lxm
```
Comment (yuluo-yx) [inline spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java:163]: 方法上可以去掉，在类上加上作者注释

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java (hunk 1 above)
Comment (fuyou-lxm) [reply spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java:51]: ok

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java (hunk 2 above)
Comment (fuyou-lxm) [reply spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java:163]: 这是record类型，实际是static class，并不是方法

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java (hunk 1 above)
Comment (fuyou-lxm) [reply spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchConfig.java]: 已删除中文注释，再Review下

File: spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java (hunk 2 above)
Comment (yuluo-yx) [reply spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java:163]: 一个类文件有一个签名或许就够了？👀

Comment (fuyou-lxm) [reply spring-ai-alibaba-core/src/main/java/com/alibaba/cloud/ai/dashscope/rag/OpenSearchVector.java]: 已修改，请再看下

**Review by yuluo-yx — APPROVED** [review]
LGTM, tks

**Pull request merged**